packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -670,151 +671,178 @@ export const useArbTokenBridge = (
-      // but we still want to detect its balance on user's wallet
-      if (isNetwork(l2ChainID).isArbitrumOne) {
-        l2Addresses.push(CommonAddress.ArbitrumOne.USDC)
-      }
-      if (isNetwork(l2ChainID).isArbitrumSepolia) {
-        l2Addresses.push(CommonAddress.ArbitrumSepolia.USDC)
-      }
+      // Callback is used here, so we can add listId to the set of listIds rather than creating a new set everytime
+      setBridgeTokens(oldBridgeTokens => {
+        const l1Addresses: string[] = []
+        const l2Addresses: string[] = []
 
-      for (const tokenAddress in bridgeTokensToAdd) {
-        const tokenToAdd = bridgeTokensToAdd[tokenAddress]
-        if (!tokenToAdd) {
-          return
+        // USDC is not on any token list as it's unbridgeable
+        // but we still want to detect its balance on user's wallet
+        if (isNetwork(l2ChainID).isArbitrumOne) {
+          l2Addresses.push(CommonAddress.ArbitrumOne.USDC)
         }
-        const { address, l2Address } = tokenToAdd
-        if (address) {
-          l1Addresses.push(address)
+        if (isNetwork(l2ChainID).isArbitrumSepolia) {
+          l2Addresses.push(CommonAddress.ArbitrumSepolia.USDC)
         }
-        if (l2Address) {
-          l2Addresses.push(l2Address)
+
+        for (const tokenAddress in bridgeTokensToAdd) {
+          const tokenToAdd = bridgeTokensToAdd[tokenAddress]
+          if (!tokenToAdd) {
+            return
+          }
+          const { address, l2Address } = tokenToAdd
+          if (address) {
+            l1Addresses.push(address)
+          }
+          if (l2Address) {
+            l2Addresses.push(l2Address)
+          }
+
+          // Add the new list id being imported (`listId`) to the existing list ids (from `oldBridgeTokens[address]`)
+          // Set the result to token added to `bridgeTokens` : `tokenToAdd.listIds`
+          const oldListIds =
+            oldBridgeTokens?.[tokenToAdd.address]?.listIds || new Set()
+          tokenToAdd.listIds = new Set([...oldListIds, listId])
         }
 
-        // Add the new list id being imported (`listId`) to the existing list ids (from `oldBridgeTokens[address]`)
-        // Set the result to token added to `bridgeTokens` : `tokenToAdd.listIds`
-        const oldListIds =
-          oldBridgeTokens?.[tokenToAdd.address]?.listIds || new Set()
-        tokenToAdd.listIds = new Set([...oldListIds, listId])
-      }
+        updateErc20L1Balance(l1Addresses)
+        updateErc20L2Balance(l2Addresses)
 
-      updateErc20L1Balance(l1Addresses)
-      updateErc20L2Balance(l2Addresses)
+        return {
+          ...oldBridgeTokens,
+          ...bridgeTokensToAdd
+        }
+      })
+    },
+    [
+      l1.network.id,
+      l1NetworkID,
+      l2.network.id,
+      updateErc20L1Balance,
+      updateErc20L2Balance
+    ]
+  )
 
-      return {
-        ...oldBridgeTokens,
-        ...bridgeTokensToAdd
-      }
-    })
-  }
+  const addBridgeTokenListToBridge = useCallback(
```

Comment:
Previously, we were passing `arbTokenBridge` to `addBridgeTokenListToBridge` https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1452/files#diff-7fe3ac4dcbf4bf9dbb24d4237a45d2c3a774f09dc34d47092793e0474eac4781L208-L220

Now, we need to refresh this when user connect, but we can't add `arbTokenBridge` to array dependency because it would cause an infinite loop